### ISSUE TITLE
Specification of CRC-16/CCITT initial value

### ISSUE THREAD

**Issue by cmcqueen** [tpircher-zz/pycrc#7]
There is some confusion about the "correct" initial value of the 16-bit CCITT CRC. See [CRC-CCITT -- 16-bit](http://web.archive.org/web/20071229021252/http://www.joegeluso.com/software/articles/ccitt.htm) ([other source](http://srecord.sourceforge.net/crc16-ccitt.html)) which explains it.

It has to do with the question of "augmented" zero bits at the end of the message, and an algorithm that can avoid the need for it, but needs a modification to the initial value. See Chapter 10, "A Slightly Mangled Table-Driven Implementation", of [A Painless Guide to CRC Error Detection Algorithms](http://www.ross.net/crc/download/crc_v3.txt) for an explanation.

In short, perhaps pycrc's `ccitt` model should have an `--xor-in` initial value of `0x1D0F` instead of `0xFFFF`.

See also:
- [CRC-16/AUG-CCITT](http://reveng.sourceforge.net/crc-catalogue/16.htm#crc.cat.crc-16-aug-ccitt) versus [CRC-16/CCITT-FALSE](http://reveng.sourceforge.net/crc-catalogue/16.htm#crc.cat.crc-16-ccitt-false) in the [Catalogue of parametrised CRC algorithms](http://reveng.sourceforge.net/crc-catalogue/)
- Online [CRC calculator](http://www.zorc.breitbandkatze.de/crc.html) which allows calculating with "nondirect" and "direct" algorithms, and converting the initial values between these two algorithms.
